tuf/ngclient/updater.py

```diff
@@ -185,8 +185,10 @@ def download_target(
         else:
             target_base_url = _ensure_trailing_slash(target_base_url)
 
-        target_filepath = targetinfo["filepath"]
         target_fileinfo: "TargetFile" = targetinfo["fileinfo"]
+        target_filepath = targetinfo["filepath"]
+        if self.config.consistent_snapshot:
+            target_filepath = target_fileinfo.hashes + "." + target_filepath
```

Comment:
`TargetFile.hashes` is a dictionary of hash_alg:hash_digest. In the legacy client we currently only download the file prefixed with the first listed hash value in [`_get_target_file()`](https://github.com/theupdateframework/tuf/blob/69dca08b9acdf58b0fa86d7789c6c56aaac1fa78/tuf/client/updater.py#L1261).

The spec describes that we should retrieve the file as _one_ of the hash digest prefixed files (see [5.7.3](https://theupdateframework.github.io/specification/latest/#fetch-target))
> Otherwise, the filename is of the form HASH.FILENAME.EXT (e.g., c14aeb4ac9f4a8fc0d83d12482b9197452f6adf3eb710e3b1e2b79e8d14cb681.foobar.tar.gz), where HASH is one of the hashes of the targets file listed in the targets metadata file found earlier in step

but that the repository should provide copies of the file at each hash digest prefix (see [6.2.1](https://theupdateframework.github.io/specification/latest/#writing-consistent-snapshots))
> consistent target files should be written to non-volatile storage as digest.filename.ext. This means that if the referrer metadata lists N cryptographic hashes of the referred file, then there must be N identical copies of the referred file, where each file will be distinguished only by the value of the digest in its filename.

Therefore the legacy implementation would be appropriate to replicate here – get the first hash from `target_fileinfo.hashes` and prepend it to `target_filepath`.